### PR TITLE
Sprint Mode 

### PR DESCRIPTION
**Pull request by samymessouak**
### **Description**  
This PR introduces a **Sprint Mode** in the challenge feature, where two players compete in a **Best of 3 (BO3)** format. The goal of this mode is for players to input as many words as possible within **1 minute per round**. The winner is determined based on the total number of words completed across all three rounds.  

**Key Features:**  
1. **Sprint Challenge Gameplay**:  
   - Each round lasts for 1 minute.  
   - The player who completes the most words after 3 rounds wins the challenge.  

2. **Automatic Challenge Cleanup**:  
   - Once the second player completes their rounds, the challenge is automatically removed from the **ongoing challenges**.  
   - Results are accessible in the **History Screen** for both players.  

3. **Database Updates**:  
   - Added two new fields to the `challenge` collection to track each player's results:  
     - `player1WordsCompleted`  
     - `player2WordsCompleted`  

4. **Navigation**:  
   - Added a route to navigate to the **Sprint Mode** screen.  

---

### **Testing**  
- Verified the functionality of the Sprint Mode rounds and result calculation.  
- Ensured navigation to Sprint Mode works as expected.  
- Confirmed challenges are deleted from the ongoing challenges after completion and appear correctly in the History Screen.

---

### **Screenshots**  
image

---

### CODE REVIEW

File: app/src/main/java/com/github/se/signify/ui/screens/challenge/SprintChallengeGameScreen.kt
Comment (ziyadbagh) [inline app/src/main/java/com/github/se/signify/ui/screens/challenge/SprintChallengeGameScreen.kt:1]: - The logic for modifying player1RoundCompleted and player2RoundCompleted in updateSprintChallenge is nearly identical and can be extracted into a helper function.
- Values like 60 for timeLeft, "completed", "in_progress", and "Loading challenge..." should be defined as constants to improve maintainability.
- Error cases (onFailure in challengeRepository.updateChallenge) are left unhandled.

**Review by ziyadbagh — CHANGES_REQUESTED** [review]
Review my comments where I’ve suggested some changes, and let me know your thoughts.

File: app/src/androidTest/java/com/github/se/signify/ui/screens/challenge/ChallengeHistoryScreenTest.kt [hunk 1]
```diff
@@ -97,15 +98,15 @@ class ChallengeHistoryScreenTest {
...
+  fun pastChallengeCard_displaysCorrectDetailsChrono() {
     val challenge =
         Challenge(
             challengeId = "challenge1",
             player1 = "testUserId",
             player2 = "opponent1",
             player1Times = mutableListOf(1000L, 2000L),
             player2Times = mutableListOf(1500L, 2500L),
-            mode = "Sprint",
+            mode = "CHRONO",
```
[the remark this answers is not in the document]
Comment (samymessouak) [reply app/src/androidTest/java/com/github/se/signify/ui/screens/challenge/ChallengeHistoryScreenTest.kt:109]: Done

File: app/src/androidTest/java/com/github/se/signify/ui/screens/challenge/SprintChallengeGameScreenTest.kt
[the remark this answers is not in the document]
Comment (samymessouak) [reply app/src/androidTest/java/com/github/se/signify/ui/screens/challenge/SprintChallengeGameScreenTest.kt:1]: Thank you

File: app/src/main/java/com/github/se/signify/MainActivity.kt
[the remark this answers is not in the document]
Comment (samymessouak) [reply app/src/main/java/com/github/se/signify/MainActivity.kt:1]: The argument is not mandatory as it is by default to null

File: app/src/main/java/com/github/se/signify/ui/screens/challenge/NewChallengeScreen.kt
[the remark this answers is not in the document]
Comment (samymessouak) [reply app/src/main/java/com/github/se/signify/ui/screens/challenge/NewChallengeScreen.kt:1]: Done.

File: app/src/main/java/com/github/se/signify/ui/screens/challenge/ChallengeHistoryScreen.kt
[the remark this answers is not in the document]
Comment (samymessouak) [reply app/src/main/java/com/github/se/signify/ui/screens/challenge/ChallengeHistoryScreen.kt:1]: Seems way better for clarity, thank you for the advice. Done.

File: app/src/main/java/com/github/se/signify/ui/screens/challenge/SprintChallengeGameScreen.kt
Comment (samymessouak) [reply app/src/main/java/com/github/se/signify/ui/screens/challenge/SprintChallengeGameScreen.kt:1]: Thank you for all of those suggestion, I added them and added some test for the helper functions.

**Review by SaFouNaldo — APPROVED** [review]
As @ziyadbagh has network problems, He asked me to approve for him.

**Review by SaFouNaldo — APPROVED** [review]

**Pull request merged**